src/tdastro/effects/redshift.py

```diff
@@ -0,0 +1,76 @@
+from tdastro.base_models import EffectModel
+
+
+class Redshift(EffectModel):
+    """A redshift effect model.
+
+    This contains a "pre-effect" method, which is used to calculate the emitted wavelengths/times
+    needed to give us the observed wavelengths and times given the redshift.
+
+    Attributes
+    ----------
+    redshift : `float`
+        The redshift.
+
+    Notes
+    -----
+    Conversions used are as follows:
+    - emitted_wavelength = observed_wavelength / (1 + redshift)
+    - emitted_time = observed_time / (1 + redshift)
+    - observed_flux = emitted_flux / (1 + redshift)
+    """
+
+    def __init__(self, redshift, **kwargs):
+        """Create a Redshift effect model.
+
+        Parameters
+        ----------
+        redshift : `float`
+            The redshift.
+        **kwargs : `dict`, optional
+           Any additional keyword arguments.
+        """
+        super().__init__(**kwargs)
+        self.redshift = redshift
+
+    def pre_effect(self, observed_times, observed_wavelengths, **kwargs):
+        """Calculate the emitted wavelengths/times needed to give us the observed wavelengths
+        and times given the redshift.
+
+        Parameters
+        ----------
+        observed_times : float
```

Comment:
This should be a numpy.ndarray so that we can vectorize over all times and wavelengths. I don't think the code needs any changes, just the comment to:

`observed_times : float or numpy.ndarray`